src/packet/ipv4.rs

```diff
@@ -76,11 +79,28 @@ pub struct Ipv4Option {
     copied: u1,
     class: u2,
     number: u5,
-    length: u8,
+    #[length_fn = "ipv4_options_length_length"]
+    length: Vec<u8>,
+    #[length_fn = "ipv4_options_payload_length"]
     #[payload]
     data: Vec<u8>,
 }
 
+fn ipv4_options_length_length(option: &Ipv4OptionPacket) -> usize {
```

Comment:
Could you add a one-line comment for this function specifying what it does?

Also, perhaps a better name would be `ipv4_option_length`?